portal_tables/unify_grant_tables.py

```diff
@@ -0,0 +1,145 @@
+        mTypes.append([m['name'] for m in syn.getChildren(grantX, includeTypes=['table'])])
+        mIds.append([m['id'] for m in syn.getChildren(grantX, includeTypes=['table'])])
+
+    # with each synapse table id mapped to a type of manifest
+    # it is possible to begin adding them to the staging table
+    # but should flatten list first for easier key-val mapping
+    flat_mTypes = list(itertools.chain(*mTypes))
+    flat_mIds = list(itertools.chain(*mIds))
+
+    idToManType = dict(zip(flat_mIds, flat_mTypes))
+    print(idToManType)
+
+    # use dictionary of synapse id to manifest type
+    # to prepare to separate data into rows
+    pubsId = []
+    datasetsId = []
+    toolsId = []
+
+    for key,val in idToManType.items():
+        if val == 'publications':
+            pubsId.append(key)
+        elif val == 'datasets':
+            datasetsId.append(key)
+        else:
+            toolsId.append(key)
+    return pubsId, datasetsId, toolsId
+
+def getEachManifestFromCCKP(syn, theNewParent):
+    """this function reads in the Synapse ID of the
+    CCKP Admin table and finds the table IDs of the
+    PublicationView, DatasetView, and ToolView manifests"""
+
+    pMan = syn.findEntityId('PublicationView', theNewParent)
+    dMan = syn.findEntityId('DatasetView', theNewParent)
+    tMan = syn.findEntityId('ToolView', theNewParent)
+
+    return pMan, dMan, tMan
+
+def main():
+    syn = synapseclient.Synapse()
+    syn.login()
+    
+    args = get_args() 
+    originTable = args.project1
+    destinationTable = args.project2
+
+    # these are the lists of each kind of manifest pulled from
+    # all individual grants
+    [allPub, allData, allTool] = getManifestsFromGrant(syn, originTable)
+    # these are the table IDs of the PublicationView, DataView, and ToolView
+    # manifests in the CCKP Admin Project table
+    [synPub, synData, synTool] = getEachManifestFromCCKP(syn, destinationTable)
+    
+    # create merged DataFrame and write to
+    # PublicationView manifest in CCKP
+    pub_dfs = []
+    for pub in allPub:
+        pub_df = syn.tableQuery(f"select * from {pub}").asDataFrame()
+        pub_dfs.append(pub_df) 
+    final_pubs = pd.concat(pub_dfs, ignore_index=True)
+    syn.store(Table(synPub, final_pubs))
+
+    # create merged DataFrame and write to
+    # DatasetView manifest in CCKP
+    dataset_dfs = []
+    for dataset in allData:
+        dataset_df = syn.tableQuery(f"select * from {dataset}").asDataFrame()
+        dataset_dfs.append(dataset_df)
+    final_datasets = pd.concat(dataset_dfs, ignore_index=True)   
+    syn.store(Table(synData, final_datasets))
+
+    # create merged DataFrame and write to
+    # ToolView manifest in CCKP
+    tool_dfs = []
+    for tool in allTool:
+        tool_df = syn.tableQuery(f"select * from {tool}").asDataFrame()
+        tool_dfs.append(tool_df)
+    final_tools = pd.concat(tool_dfs, ignore_index=True)
+    syn.store(Table(synTool, final_tools))
```

Comment:
This chunk of code is similar between the three View tables.  I recommend extracting it into a function in that case.